openid/httpclientmock_test.go

```diff
@@ -40,14 +41,17 @@ type decodeResponseResp struct {
 	err   error
 }
 
-func (c *HTTPClientMock) httpGet(r *http.Request, url string) (*http.Response, error) {
-	c.Calls <- &httpGetCall{url}
+func (c *HTTPClientMock) httpGet(req *http.Request, url string) (*http.Response, error) {
+	c.Calls <- &httpGetCall{req, url}
 	gr := (<-c.Calls).(*httpGetResp)
 	return gr.resp, gr.err
 }
 
-func (c *HTTPClientMock) assertHttpGet(url string, resp *http.Response, err error) {
+func (c *HTTPClientMock) assertHttpGet(req *http.Request, url string, resp *http.Response, err error) {
 	call := (<-c.Calls).(*httpGetCall)
+	if req == nil || call.req != req {
```

Comment:
similar to below where we have != anything this should be != nil this way for the tests where you don't care for asserting the request instance you can just continue to pass nil